core/filter_api.php

```diff
@@ -3434,3 +3418,202 @@ function filter_print_view_type_toggle( $p_url, $p_view_type ) {
+			error_parameters( $p_filter_id );
+			trigger_error( ERROR_FILTER_NOT_FOUND, ERROR );
+		} else {
+			return $p_default;
+		}
+	}
+	$t_filter = filter_deserialize( $t_filter_string );
+	$t_filter = filter_clean_runtime_properties( $t_filter );
+	$t_filter['_filter_id'] = $p_filter_id;
+
+	$t_filter = filter_update_source_properties( $t_filter );
+
+	return $t_filter;
+}
+
+function filter_update_source_properties( array $p_filter ) {
+	# Check if the filter references a named filter
+	# This property only makes sense, and should be available on unnamed filters
+	# Note that an unnamed filter is used as a working copy for the user. This logic
+	# allows for automatic updating of a current filter when the user had selected
+	# a named filter and that filter has changed. Otherwise the updates in base filter
+	# wuould not reflect in the user's current filter
+	if( isset( $p_filter['_filter_id'] ) ) {
+		$t_filter_id = $p_filter['_filter_id'];
+	} else {
+		$t_filter_id = null;
+	}
+	if( isset( $p_filter['_source_query_id'] ) && $t_filter_id != $p_filter['_source_query_id'] ) {
+		$t_source_query_id = $p_filter['_source_query_id'];
+		# check if filter id is a proper named filter, and is accesible
+		if( filter_is_named_filter( $t_source_query_id ) && filter_is_accessible( $t_source_query_id ) ){
+			# replace filter with the referenced one
+			$t_new_filter = filter_deserialize( filter_db_get_filter_string( $t_source_query_id ) );
+			# update the referenced stored filter id for the new loaded filter
+			$t_new_filter['_source_query_id'] = $t_source_query_id;
+			$p_filter = filter_copy_runtime_properties( $p_filter, $t_new_filter );
+		} else {
+			# If the filter id is not valid, clean the referenced filter id
+			unset( $p_filter['_source_query_id'] );
+		}
+	}
+	return $p_filter;
+}
+
+/**
+ * Returns a filter which is stored in session data, indexed by the provided key.
+ * A default value can be provided to be used when the key doesn't exists
+ *
+ *  You may pass in any array as a default (including null) but if
+ *  you pass in *no* default then an error will be triggered if the key
+ *  cannot be found
+ *
+ * @param string $p_filter_key  Key to look up for in session data
+ * @param mixed $p_default		A default value to return if key not found
+ * @return array	A filter array.
+ */
+function filter_temporary_get( $p_filter_key, $p_default = null ) {
+	# if no default was provided, we will trigger an error if not found
+	$t_trigger_error = func_num_args() == 1;
+
+	$t_session_filters = session_get( 'temporary_filters', array() );
+	if( isset( $t_session_filters[$p_filter_key] ) ) {
+		# setting here the key in the filter array only if the key exists
+		# this validates against receiving garbage input as XSS attacks
+		$t_filter = $t_session_filters[$p_filter_key];
+		$t_filter['_temporary_key'] = $p_filter_key;
+		return filter_ensure_valid_filter( $t_filter );
+	} else {
+		if( $t_trigger_error ) {
+			error_parameters( $p_filter_id );
```

Comment:
OK, the review here is anchored at filter_get(), that why i was confused.
But you mean in filter_temporary_get(). I got it.